api/src/controllers/document-controller.js

```diff
@@ -254,15 +254,15 @@ exports.protectedDelete = async function(args, res, next) {
  * @param {*} [url=null] url of the document
  * @param {*} [readRoles=[]] read roles to add to the document. (optional)
  */
-async function createURLDocument(fileName, addedBy, url, readRoles = []) {
+async function createURLDocument(fileName, addedBy, url, readRoles = ['admin:lng', 'admin:nrced', 'admin:bcmi']) {
```

Comment:
Currently, the readRoles array only is used to set `public`, so these defaults will always get overridden.

I think we would just want to set those 3 roles directly in the document every time, in the same way the write roles are set.

Reply:
`  document.read = ['sysadmin', 'admin:lng', 'admin:nrced', 'admin:bcmi', ...readRoles];`

Reply:
Assuming that my blub about documents needing all roles is accurate, and i overlooked anything.

Reply:
As for the code change above, no default constructor variables needed - so remove the auto-assignment.  Leave `readRoles = []` and add `writeRoles =[]`.

As for the comment above, we shouldn't hard code roles inline like that.  Let the caller of the function define what they should be so there's less automatic behaviors we might need to unwind later if business rules change.

```document.read = ['sysadmin', ...readRoles];```
and
```document.write= ['sysadmin', ...writeRoles];```